bytecomp/matching.ml

```diff
@@ -1576,11 +1586,13 @@ let divide_tuple arity p ctx pm =
 
 (* Matching against a record pattern *)
 
-
 let record_matching_line num_fields lbl_pat_list =
-  let patv = Array.make num_fields omega in
-  List.iter (fun (_, lbl, pat) -> patv.(lbl.lbl_pos) <- pat) lbl_pat_list;
-  Array.to_list patv
+  try
+    let patv = Array.make num_fields omega in
+    List.iter (fun (_, lbl, pat) -> patv.(lbl.lbl_pos) <- pat) lbl_pat_list;
+    Array.to_list patv
+  with Invalid_argument _ -> assert false
```

Comment:
It's not clear at all, to me, where this `Invalid_argument _` exception would be raised. If you mean that `lbl.lbl_pos` is always a valid index for `patv`, why don't you just write `assert (lbl.lbl_pos < Array.length patv)` in the loop?